### ISSUE TITLE
Postgres error: relation "x" already exists

### ISSUE THREAD

**Issue by eiskalteschatten** [strapi/strapi#5113]
**Describe the bug**
When starting Strapi with content types that have relationships to other content types, I get the following stack trace:

```
error error: relation "components_distribution_page_distribution_page_languages_compon" already exists
at Connection.parseE (/app/node_modules/pg/lib/connection.js:604:13)
at Connection.parseMessage (/app/node_modules/pg/lib/connection.js:403:19)
at TLSSocket.<anonymous> (/app/node_modules/pg/lib/connection.js:123:22)
at TLSSocket.emit (events.js:321:20)
at addChunk (_stream_readable.js:297:12)
at readableAddChunk (_stream_readable.js:273:9)
at TLSSocket.Readable.push (_stream_readable.js:214:10)
at TLSWrap.onStreamRead (internal/stream_base_commons.js:186:23)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! strapi@0.1.0 start: `strapi start`
npm ERR! Exit status 1
```

The error doesn't occur locally when I use a sqlite database. It only seems to occur with Postgres.

**Steps to reproduce the behavior**
1. Set up Strapi with a Postgres database
2. Create two content types
3. Create a relationship between the content types
4. Manually stop the server
5. Manually start the server (it seems to have no problem when the server automatically restarts after changing content types using the admin interface)

**Expected behavior**
Strapi should start normally. The table obviously exists and should not have to be dropped just to get Strapi started.

**System**
- Node.js version: 10.18.1
- NPM version: 6.13.4
- Strapi version: 3.0.0-beta.18.6
- Database: Postgres
- Operating system: macOS (local) and Linux (server) - happens on both systems

**Comment by derrickmehaffy** [comment]
I was unable to reproduce this myself. @eiskalteschatten can you upload an example project with a dump of the PG database to test with?

**Comment by eiskalteschatten** [comment]
No problem. Here are my api and components folders as well as a dump of the mostly empty test DB.
[strapi.sql.zip](https://github.com/strapi/strapi/files/4122779/strapi.sql.zip)
[api-and-components.zip](https://github.com/strapi/strapi/files/4122780/api-and-components.zip)

**Comment by eiskalteschatten** [comment]
After playing around with it some more, I think I've figured out what the basic problem is. I get this error every time the table name is too long to fit the normal naming scheme that Strapi uses to name its tables.

In the example above, the table name "components_distribution_page_distribution_page_languages_compon" is cut off. Once I renamed the relation so that it was "components_distribution_page_dist_page_languages_components", then it worked perfectly fine.
